packages/amplify_authenticator/lib/amplify_authenticator.dart

```diff
@@ -557,54 +726,103 @@ class _AuthenticatorBody extends StatelessWidget {
-            child: Navigator(
-              onPopPage: (_, dynamic __) => true,
-              pages: [
-                if (authenticatorScreen == null)
-                  MaterialPage<void>(
-                    child: Theme(
-                      data: userAppTheme,
-                      child: child,
-                    ),
-                  ),
-                if (authenticatorScreen != null)
-                  MaterialPage<void>(
-                    child: ScaffoldMessenger(
-                      key: _AuthenticatorState.scaffoldMessengerKey,
-                      child: Scaffold(
-                        backgroundColor:
-                            AmplifyTheme.of(context).backgroundPrimary,
-                        body: SizedBox.expand(
-                          child: authenticatorScreen is AuthenticatorScreen
-                              ? SingleChildScrollView(
-                                  child: authenticatorScreen)
-                              : authenticatorScreen,
-                        ),
-                      ),
-                    ),
-                  ),
-              ],
-            ),
+            child: builder(authState, authenticatorScreen),
           );
         },
       ),
     );
   }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(
+        ObjectFlagProperty<Widget Function(AuthState state, Widget child)>.has(
+            'builder', builder));
+  }
+}
+
+// The body of the authenticator
+class _AuthenticatorBody extends StatelessWidget {
+  const _AuthenticatorBody({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return _AuthStateBuilder(
+      child: child,
+      builder: (state, child) {
+        return Navigator(
+          onPopPage: (_, dynamic __) => true,
+          pages: [
+            if (state is AuthenticatedState) MaterialPage<void>(child: child),
+            if (state is! AuthenticatedState)
+              MaterialPage<void>(
+                child: ScaffoldMessenger(
+                  key: _AuthenticatorState.scaffoldMessengerKey,
+                  child: Scaffold(
+                    backgroundColor: AmplifyTheme.of(context).backgroundPrimary,
+                    body: SizedBox.expand(
+                      child: child is AuthenticatorScreen
+                          ? SingleChildScrollView(child: child)
+                          : child,
+                    ),
+                  ),
+                ),
+              ),
+          ],
+        );
+      },
+    );
+  }
+}
+
+/// A widget for wrapping portions of the application (typically routes)
+/// that are intended to be protected by authentication.
+class AuthenticatedView extends StatelessWidget {
+  const AuthenticatedView({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
```

Comment:
So it shows on the ctor as well

```suggestion
/// {@template amplify_authenticator.authenticated_view}
/// A widget for wrapping portions of the application (typically routes)
/// that are intended to be protected by authentication.
/// {@endtemplate}
class AuthenticatedView extends StatelessWidget {
  /// {@macro amplify_authenticator.authenticated_view}
  const AuthenticatedView({
    Key? key,
    required this.child,
  }) : super(key: key);
```